### DISCUSSION TITLE
Attaching files to bot-based message extensions plugins for Copilot in TEAMS

### DISCUSSION THREAD

**Discussion by aymiee** [OfficeDev/microsoft-365-agents-toolkit#11918] [Q&A]
According to Microsoft, only bot-based message extensions with search commands can be extended as plugins for Copilot for Microsoft 365. However, I'm looking for ways to attach files directly to Copilot within Microsoft Teams and interact using a message extension plugin. Currently, attachment seems limited to OneDrive/SharePoint, and using the Graph API to retrieve files by name risks fetching duplicates with incorrect content. Does anyone have alternative solutions or insights on how to set this up more reliably

**Comment by davrous** [comment]
Hello! We will soon open API Plugins as a Copilot extensibility option but you're right, for now, only bot based message extensions can work as a plugin for Copilot.

I'm not sure I understood what you're trying to do. Can you outline more in details what is the expected user experience you'd like to build using a Copilot plugin?

**Reply by aymiee** [reply]
Hi @davrous I am using Copilot in Teams to summarize the contents of Excel files, which currently undergo a manual data transformation process to convert them to text. Since uploading text files directly is not supported, I manually copy this text into Copilot for processing. I would like to automate this step. Ideally, the user would upload an Excel file directly into Copilot via a message plugin, which would then convert the Excel file to text and automatically input the converted text into Copilot for summarization. Currently, our Copilot (in Teams) supports uploading only Office document types like Word and Excel.

There's also the limitation in retrieving attachments directly via the handleTeamsMessagingExtensionQuery, which is an essential capability for automating this workflow. I am seeking solutions to overcome this retrieval issue as part of the automation process.

Will the API plugins as a Copilot extensibility option support file upload?
